### PR TITLE
Add request argument to TemplateResponse (v1.x)

### PR DESCRIPTION
**Pull request by alex-oleshkevich**
This PR removes backward compatibility done in https://github.com/encode/starlette/pull/2191
To be merged after 1.0 release.

### CODE REVIEW

File: starlette/templating.py [hunk 1]
```diff
@@ -142,17 +142,16 @@ def get_template(self, name: str) -> "jinja2.Template":
 
     def TemplateResponse(
         self,
+        request: Request,
```
Comment (aminalaee) [inline starlette/templating.py:145]: Any reason why this is not an optional argument so it can be released without breaking?

**Comment by Kludex** [comment]
I meant to create a PR so I could see the diff between the PR that adds deprecation and what is supposed to be refactor for v1.

I mean, target the PR you are building from this one.

**Comment by alex-oleshkevich** [comment]
@Kludex this PR is the target repo state after removing all deprecations.

**Comment by Kludex** [comment]
Screenshot 2023-07-03 at 17 16 06

I need instead of "encode:master" to be "alex-oleshkevich:templates-request". 🤔

**Comment by alex-oleshkevich** [comment]
See it here https://github.com/alex-oleshkevich/starlette/pull/56/files

**Comment by Kludex** [comment]
> To be merged after 1.0 release.

Not after, but for 1.0.

File: starlette/templating.py (hunk 1 above)
Comment (Kludex) [reply starlette/templating.py:145]: This was discussed on https://github.com/encode/starlette/pull/2191/.

The diff here is actually incomplete.

**Comment by Kludex** [comment]
I've applied this on https://github.com/encode/starlette/pull/2384, and given co-author to @alex-oleshkevich . FYI

**Pull request closed without merging**